### DISCUSSION TITLE
Understanding "response = response.rt == null ? info : response;"

### DISCUSSION THREAD

**Discussion by SHogenboom** [jspsych/jsPsych#1650] [Q&A] [answered]
I'm building my own plugin where, among other things, I want to record each key press (rt, key) while someone is typing. Looking through all the plugins I imagined that `jspsych-serial-reaction-time.js` would be good place to start and saw that a key press results in the `after_response` function being triggered:

```
// function to handle responses by the subject
    function after_response(info) {

     // only record first response
      response = response.rt == null ? info : response;

      if (trial.response_ends_trial) {
        endTrial();
      }
    };
};
```

I'm left baffled by the `response = response.rt == null ? info : response;` - but my gut feeling says that this is what I need to use. Can someone explain to me how the code works so I know how to use/adjust it in my experiment? 

Thanks!

**Comment by SHogenboom** [comment]
The awesome tutorial video (which I hadn't watched entirely yet) obviously shows that the code above is not necessary for what I want to achieve:
https://www.youtube.com/watch?v=XQcsFwAmbiw

An answer to understand what is happening would still be appreciated though :)

**Comment by jodeleeuw** [comment, marked as answer]
Hi @SHogenboom. I moved this over to Discussion from Issues.

```js
response = response.rt == null ? info : response;
```

This is a shorthand for an `if... else` statement. It's called the ternary operator. You can learn more about it [here](https://www.javascripttutorial.net/javascript-ternary-operator/).

**Reply by SHogenboom** [reply]
Thanks! That clarifies a lot :)
